integration-tests/src/test/kotlin/org/stellar/anchor/platform/test/Sep24End2EndTests.kt

```diff
@@ -100,27 +99,25 @@ class Sep24End2EndTest(config: TestConfig, val jwt: String) {
       assertEquals(fetchedTxn.id, transactionByStellarId.id)
 
       // Check the events sent to the reference server are recorded correctly
-      val actualEvents = waitForBusinessServerEvents(response.id, 4)
-      assertNotNull(actualEvents)
-      actualEvents?.let { assertEquals(4, it.size) }
+      val actualEvents = anchorReferenceServerClient.pollEvents(response.id, 4)
       val expectedEvents: List<SendEventRequest> =
         gson.fromJson(
           expectedDepositEventsJson,
           object : TypeToken<List<SendEventRequest>>() {}.type
         )
-      compareAndAssertEvents(asset, expectedEvents, actualEvents!!)
+      compareAndAssertEvents(asset, expectedEvents, actualEvents)
 
       // Check the callbacks sent to the wallet reference server are recorded correctly
-      val actualCallbacks = waitForWalletServerCallbacks(response.id, 4)
-      actualCallbacks?.let {
-        assertEquals(4, it.size)
-        val expectedCallbacks: List<Sep24GetTransactionResponse> =
-          gson.fromJson(
-            expectedDepositCallbacksJson,
-            object : TypeToken<List<Sep24GetTransactionResponse>>() {}.type
-          )
-        compareAndAssertCallbacks(asset, expectedCallbacks, actualCallbacks)
-      }
+      val actualCallbacks =
+        walletServerClient.pollCallbacks(response.id, 4).map {
+          gson.fromJson(it, Sep24GetTransactionResponse::class.java)
```

Comment:
Can you add some comments of why removing `assertEquals(4, it.size)` would fix the problem?

Reply:
It doesn't fix the problem. The assertion was required before because `waitForWalletServerCallbacks` returned a null if the response size is not 4. I changed the API of `waitForWalletServerCallbacks` / `pollCallbacks` so that it will always return a list, so the assertion can be removed.